code/DepMapGeneEss/Instructions.txt

```diff
@@ -0,0 +1,27 @@
+1. First  make sure you have the DepMap data - we currently use DepMap version 2021Q3. You need to go to the downloads section at
+https://depmap.org/portal/download/all/ , select 2021Q3.
+Download and place the files Achilles_gene_effect.csv and CCLE_expression_full.csv in the Human-GEM/code/DepMapGeneEss/data folder.
+2. Run the R script ConvertGeneEssInputData.R. Note that you need to adjust the folder - R is hopeless when it comes to 
+relative paths from the file - the information just doesn't exist. This will generate the file DepMap_tpm_gene_symbols.txt
+3. Run PrepDepMapData.m - you may have to modify the folder here as well. This will take an hour or so and generates the files
+arrayDataDepMap.mat and prepDataGeneSymbols.mat.
+4. Create a folder on the cluster with a suitable name, for example "GeneEssDepMap". Within that folder, create a folder named "components".
+Copy RAVEN, COBRA, and Human-GEM to components, they need to reside under those exact names, because those and underlying folders will be
+added to the matlab path. Make sure the Human-GEM folder Human-GEM/code/DepMapGeneEss/data is copied as well with
+the files that we generated above.
+5. Go back to your top folder ("GeneEssDepMap"). Create a folder DepMapRuns. Copy the files run_gen_depmap_models.sh, 
+run_evaless_depmap.sh, generate_DepMap_models_ftINIT_Cluster.m, and getTaskEssentialGenesCluster.m to that folder. Also create a folder 
+called "logs" in that folder.
+6. Make sure you are in the DepMapRuns folder. First, generate the models using the following line (also available in the .sh script):
+sbatch -o logs/run_gen_depmap_models-%A-%a.log --array=1-40 run_gen_depmap_models.sh
+It may be needed to init cobra and init RAVEN (SetRavenSolver etc.). TODO: Describe this here.
+7. Run sbatch -o logs/run_evaless_depmap-%A-%a.log --array=1-40 run_evaless_depmap.sh after the model generation has completed
+8. Download all produced files to the data folder on your computer. These are found in components/Human-GEM/code/DepMapGeneEss/data
+9. Run evaluateDepMapEssentialityPredictions.m. Note that you may need to change the folder in the file and that there are some things in 
+the file that need to be modified to match what you want to do. The file currently only gets stats for one model version, but can easily be expanded
+to take in data from more model versions, although these also need to be processed as above as well.
+10. Run PlotGeneEss.R in R to make a violin plot of MCC for each model. The code needs some modification to fix paths and number of models.
```

Comment:
How about moving this to the Human-GEM guide?

Reply:
I don't know what is best. The important thing is to have it somewhere.

Reply:
Having this as atutorial

> Great job! I've left some aesthetic comments.
> 
> In terms of the bigger picture, what is the aim here? To be able to run this once, now and again? Or more often, say with every other release?

I would say with the larger essentiality check for DepMap data should be performed every major release in the main branch. But for each PR, then we can run a smaller scale test as Hao committed in #563

Reply:
Here's my thinking: right now I am strongly in favour of moving it to the guide, probably as new file `gem_validation`. The purpose there would be that this documentation would be applicable to other datasets as well. And in the spirit of reproducibility, my thinking is that this documentation should be out in the open, instead of buried in a subfolder. There is also no equivalent of this documented somewhere else, so it's adding value.
I'm also imagining the PR that adds this to the guide to be marked as `draft` for a while, until the documentation gets refined enough.

What do you and other think?

Reply:
Pinging @haowang-bioinfo and @feiranl in particular.

Reply:
will think about it

Reply:
> the scope of this PR appears to have more generic usage, and may suit a standalone repo for specialised development?

If not Human-GEM, then maybe RAVEN?